src/main/java/io/cryostat/net/reports/ActiveRecordingReportCache.java

```diff
@@ -177,4 +189,32 @@ protected String getReport(RecordingDescriptor recordingDescriptor, String filte
             }
         }
     }
+
+    @Override
+    public Set<String> categories() {
+        return Set.of(STOP_NOTIFICATION_CATEGORY);
+    }
+
+    @Override
+    public void onMessage(String category, Message<?> message) {
+        Map<String, Object> notification = gson.fromJson(message.body().toString(), Map.class);
+        Map<String, Object> msg = objectMapper.convertValue(notification.get("message"), Map.class);
+        Map<String, Object> recording = objectMapper.convertValue(msg.get("recording"), Map.class);
```

Comment:
This is unexpected. Why does it need both the Jackson ObjectMapper and Gson? Both of those fill the same role of (de)serializing POJO to/from JSON.

Reply:
> > Not sure how to bind the ActiveRecordingReportCache into the set of listeners in the NotificationPublisher without making it public.
> 
> Not sure exactly what you mean here.
> 
> https://github.com/cryostatio/cryostat/blob/4c45fbc451e2c41deadd09679e8bbc98d5f4849f/src/main/java/io/cryostat/net/web/http/api/v2/HttpApiV2Module.java#L94
> 
> https://github.com/cryostatio/cryostat/blob/4c45fbc451e2c41deadd09679e8bbc98d5f4849f/src/main/java/io/cryostat/net/web/http/api/v2/ApiGetHandler.java#L62
> 
> Does this example help answer the question at all?
> 

Sorry, I meant that the `NotificationsModule` is in `java.io.cryostat.messaging.notifications.NotificationsModule` and the report cache is in `java..io.cryostat.net.reports.ActiveRecordingReportCache` so I think the `NotificationsModule` can't see the cache as a class since it's in a different package and that's what I meant. 

> > Also may be better ways to parse the Notification message but I don't think using POJOs here is smart
> 
> Why?

Uh, I think I was thinking like generally, since Notifications have different structures, it may be difficult to extend the parsing to all kinds, but I realize that Notifications are all generally the same structure of `Meta` and `Message` so it probably isn't that hard to parse.